### ISSUE TITLE
"Stock status" not global

### ISSUE THREAD

**Issue by webrod** [magento/magento2#32923]
### Preconditions (*)

1. Magento 2.4.2
2. I have only 1 store. so there is store_id=0 (admin) and store_id=1 (my website)

### Steps to reproduce (*)

1. change "Stock status" from a product (e.g. from 'out of stock' to 'in stock')
2.

### Expected result (*)

1. status has changed for the 2 stores since it says it is a global value (both admin and frontend should be updated)
![image](https://user-images.githubusercontent.com/1897537/116781056-c73afe00-aa80-11eb-86b3-b1375893f976.png)

### Actual result (*)

1. only admin store has been updated. so only admin says 'in stock', so the frontend always shows 'out of stock' 
2. 

![image](https://user-images.githubusercontent.com/1897537/116780922-d3728b80-aa7f-11eb-985d-3053751c5617.png)

 ---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

my workaround is to update cataloginventory_stock_item in DB.
I cannot even switch to my frontend view from admin, the list to pick a different store is hidden, probably because there is only one store. 
![image](https://user-images.githubusercontent.com/1897537/116780994-4845c580-aa80-11eb-952c-5c479d1f1d59.png)

also why does it manage a stock per website? especially one stock for admin and one for frontend? can I disable this in case it helps? 
thanks
Rod

**Comment by webrod** [comment]
@magento give me 2.4-develop instance

**Comment by webrod** [comment]
interesting, on the magento instance, I was able to switch from "all store views"/"default store view".
and "stock status" IS global here, whether I change on either view, it applies to all, I can see that by switching the view after an update.
But on my website, fresh magento+ data/config migration from M1.9 to M2.4, this attribute is per store. I was able to change the setting "enable single store view' so now I can switch between the views and easily confirm updates of "Stock status" attribute are NOT global. 
very weird.

probably the migration has broken something.
any idea how to bring back this field to GLOBAL?

thanks,
Rod
